panels/sharing/cc-sharing-panel.c

```diff
@@ -646,6 +660,154 @@ cc_sharing_panel_setup_media_sharing_dialog (CcSharingPanel *self)
                                            WID ("media-sharing-status-label"));
 }
 
+static void
+content_sharing_switch_active_changed_cb (GtkSwitch *widget,
+                                          GParamSpec *pspec,
+                                          CcSharingPanel *self)
+{
+  GError *error = NULL;
+
+  if (gtk_switch_get_active (widget))
+    {
+      g_dbus_proxy_call_sync (self->priv->companion_app_avahi_helper_proxy,
+                              "EnterDiscoverableMode",
+                              g_variant_new ("(i)", -1, "Endless Computer"),
```

Comment:
If anything, this should be `g_variant_new ("(s)", "Endless Computer")`, but I think we should just get rid of the name parameter, as mentioned in the other PR.

Reply:
Note that if we do that, it will also be much easier to either make `Discoverable` a writable property, or to provide a `SetDiscoverable()` method that takes a boolean, instead of two separate ones.

Reply:
Ouch, yeah, good catch on the arguments.

My preference here would be to have a single atomic operation that sets up discoverability, and configures which name should be used (since we will have different names), rather than having multiple properties over the place. That is a personal design preference of mine, so I'm willing to override it if there's a good reason to have multiple properties.

Reply:
I'm going to merge this as-is and offer a suggestion for a better approach in T20668.